### PR TITLE
[docs] Make it clear to "fork" Ray repo

### LINKED ISSUE #39972: [docs] Emphasize need to fork the repo for docs contributors
### Description

Readers miss the instruction for forking the repo in the doc for setting up the local docs build. Let's make it a little clearer and link to instructions. Better yet, @peytondmurray had a great suggestion to include a picture of the fork button.

### Link

https://docs.ray.io/en/latest/ray-contribute/docs.html
https://docs.ray.io/en/master/ray-contribute/development.html#clone-the-repository

### PR DESCRIPTION
**Pull request by emmyscode**
## Why are these changes needed?

This PR targets two pages in the docs related to building Ray, and specifically emphasizes the need to fork the repository first rather than cloning straightaway. I have also included a few changes related to the style and readability of relevant sections affected. These changes hopefully cut down on the amount of text while preserving usefulness.

## Related issue number

Closes: #39972

cc: @angelinalg 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-contribute/development.rst [hunk 1]
```diff
@@ -14,103 +14,58 @@ For a majority of Ray users, installing Ray via the latest wheels or pip package
...
 
-Clone the repository
+Fork the Ray repository
 --------------------
 
-To build Ray locally you will need to have the Git repository, so first, fork it on GitHub. Then you can clone it to your machine:
+Forking an open source repository is a best practice when looking to contribute, as it allows you to make and test changes without affecting the original project, ensuring a clean and organized collaboration process.
+You can propose changes to the main project by submitting a pull request to the main project's repository.
 
-.. tab-set::
```
Comment (emmyscode) [inline doc/source/ray-contribute/development.rst:22]: This removal may be slightly controversial, but my line of thought here is that the GitHub documentation is so clear, and it effectively covers Mac, Windows, Linux, GitHub CLI, Desktop, and Web browser methods. In contrast, the portion I removed only covers one method of cloning instructions and doesn't address forking. Why try to provide an in-place watered down version of really really well-done GitHub docs?

File: doc/source/ray-contribute/docs.ipynb [hunk 2]
```diff
@@ -32,7 +32,9 @@
     "If you want to contribute to the Ray documentation, you'll need a way to build it.\n",
     "You don't have to build Ray itself, which is a bit more involved.\n",
     "\n",
-    "You just have to [clone the Ray repository](https://docs.ray.io/en/master/ray-contribute/development.html#clone-the-repository), then [prepare the Python environment](https://docs.ray.io/en/master/ray-contribute/development.html#prepare-the-python-environment).\n",
+    "Follow the instructions in [Building Ray from Source](https://docs.ray.io/en/master/ray-contribute/development.html#building-ray-from-source) up to and including the following:\n",
+    "1. [Fork the Ray repository](https://docs.ray.io/en/master/ray-contribute/development.html#fork-the-ray-repository)\n",
+    "2. [Prepare a Python Virtual Environment](https://docs.ray.io/en/master/ray-contribute/development.html#prepare-a-python-virtual-environment)\n",
```
Comment (peytondmurray) [inline doc/source/ray-contribute/docs.ipynb:37]: If we're going to be following the same steps as the section above above, I think this should be step 3, with step 2 being "Clone the repo."

**Review by peytondmurray — APPROVED** [review]
Minor suggestion, otherwise this looks good to me :rocket:

File: doc/source/ray-contribute/development.rst [hunk 3]
```diff
@@ -14,103 +14,58 @@ For a majority of Ray users, installing Ray via the latest wheels or pip package
 .. contents::
   :local:
 
-Clone the repository
+Fork the Ray repository
 --------------------
```
Comment (peytondmurray) [inline doc/source/ray-contribute/development.rst:18]: This line is breaking the RTD build. Extend the `-` to match the line length of the title above.

File: doc/source/ray-contribute/docs.ipynb (hunk 2 above)
Comment (emmyscode) [reply doc/source/ray-contribute/docs.ipynb:37]: To clarify, you're proposing the flow to be:
1. Fork the repo
2. Clone the repo
3. Prepare a Python virtual environment

I have these two steps (fork, prepare) to correspond to the two sections in the "Building Ray from Source" doc.

File: doc/source/ray-contribute/development.rst [hunk 4]
```diff
@@ -14,103 +14,58 @@ For a majority of Ray users, installing Ray via the latest wheels or pip package
...
-
-            git clone https://github.com/[your username]/ray.git
-
-Then you can enter into the Ray git repository directory:
-
-.. code-block:: shell
+1. Navigate to the `Ray GitHub repository <https://github.com/ray-project/ray>`_.
+2. Follow these `GitHub instructions <https://docs.github.com/en/get-started/quickstart/fork-a-repo>`_, and do the following:
+    
+    a. `Fork the repo <https://docs.github.com/en/get-started/quickstart/fork-a-repo#forking-a-repository>`_ with your preferred method.
```
Comment (angelinalg) [inline doc/source/ray-contribute/development.rst:26]:
```suggestion
    a. `Fork the repo <https://docs.github.com/en/get-started/quickstart/fork-a-repo#forking-a-repository>`_ using your preferred method.
```

File: doc/source/ray-contribute/docs.ipynb [hunk 5]
```diff
@@ -32,7 +32,10 @@
     "If you want to contribute to the Ray documentation, you'll need a way to build it.\n",
```
Comment (angelinalg) [inline doc/source/ray-contribute/docs.ipynb:32]:
```suggestion
    "If you want to contribute to the Ray documentation, you need a way to build it.\n",
```

**Review by angelinalg — APPROVED** [review]
These are great improvements. Thank you for doing this!

File: doc/source/ray-contribute/docs.ipynb [hunk 6]
```diff
@@ -32,7 +32,10 @@
...
     "You don't have to build Ray itself, which is a bit more involved.\n",
     "\n",
-    "You just have to [clone the Ray repository](https://docs.ray.io/en/master/ray-contribute/development.html#clone-the-repository), then [prepare the Python environment](https://docs.ray.io/en/master/ray-contribute/development.html#prepare-the-python-environment).\n",
+    "Follow the instructions in [Building Ray from Source](https://docs.ray.io/en/master/ray-contribute/development.html#building-ray-from-source) up to and including the following:\n",
+    "1. [Fork the Ray repository](https://docs.ray.io/en/master/ray-contribute/development.html#fork-the-ray-repository)\n",
+    "2. [Clone the repository](https://docs.ray.io/en/master/ray-contribute/development.html#fork-the-ray-repository) to your local machine\n",
```
[the remark this answers is not in the document]
Comment (emmyscode) [reply doc/source/ray-contribute/docs.ipynb]: Doesn't hurt to be more clear!

**Comment by angelinalg** [comment]
@emmyscode Would you mind rebasing and if tests pass, please ask @can-anyscale for help merging.

**Pull request merged**